File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/user/dto/WithdrawalStatusResponse.java
package com.portfolio.auctionmarket.domain.user.dto;

import com.portfolio.auctionmarket.domain.auctions.entity.Auction;
import com.portfolio.auctionmarket.domain.bids.entity.Bid;
import com.portfolio.auctionmarket.domain.user.entity.User;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class WithdrawalStatusResponse {

    private String nickname;
    private Long currentPoint;
    private Long bidCount;
    private Long productCount;

    public static WithdrawalStatusResponse from(User user, Long bidCount, Long productCount) {
        return WithdrawalStatusResponse.builder()
                .nickname(user.getNickname())
                .currentPoint(user.getPoint())
                .bidCount(bidCount)
                .productCount(productCount)
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/user/dto/WithdrawalStatusResponse.kt
package org.example.auctioncommon.domain.user.dto

import org.example.auctioncommon.domain.user.entity.User

data class WithdrawalStatusResponse(
    val nickname: String,
    val currentPoint: Long,
    val bidCount: Long,
    val productCount: Long,
)

fun User.toWithdrawalStatusDto(bidCount: Long, productCount: Long) : WithdrawalStatusResponse{
    return WithdrawalStatusResponse(
        nickname = this.nickname,
        currentPoint = this.point,
        bidCount = bidCount,
        productCount = productCount
    )
}
